### PR TITLE
3.9'ify Friendo

### LINKED ISSUE #208: 3.9'ify Friendo_Bot
We should be able to bring Friendo up to python 3.9. The only changes to be made should be updating the piplock and updating the docker. We should maintain existing typing as 3.8 and continue to unless there is another reason for Friendo to not be compatible with 3.8.

### PR DESCRIPTION
**Pull request by KJ002**
This pr updates Friendo up to `python3.9`. This mainly alters the docker file, notably adding `gcc` to the library installs because there is now a sub-dependency as of this new version that must be compiled. 

closes #208

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version: "3.7"
+version: "3.9"
```
Comment (fisher60) [inline docker-compose.yml:1]: This is the docker-compose version. 3.9 is not released yet. This should be left as 3.7 for now.

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
-FROM python:3.8-slim
+FROM python:3.9-slim
 
 # Install dependencies for OpenCV
-RUN apt-get update && apt-get -y install libgl1 libglib2.0-0 git
+RUN apt-get update && apt-get -y install libgl1 libglib2.0-0 git gcc
```
Comment (fisher60) [inline Dockerfile:4]: We do not want to add gcc as part of the build process. Can you find a different way to accomplish updating packages for 3.9?

**Review by fisher60 — CHANGES_REQUESTED** [review]
Hello, thank you for the PR. There are a few issues I see here that are noted as requested changes. Feel free to discuss them in the comments.

Your commit messages are quite short and uninformative. Many of these should have a commit body and probably a bit longer of a title. I.e multiple `update 3.9` commits tells me nothing, especially since this entire pr is `update 3.9`. Furthermore I needed to assume you are adding gcc as a dependency due to some packages not having build wheels for 3.9, this would have been a perfect thing to add to the commit body.

File: Dockerfile (hunk 2 above)
Comment (ChrisLovering) [reply Dockerfile:4]: This whole line can be removed (no `apt` packages in the docker image) by bumping d.py to 1.7 and changing to `opencv-python-headless`.

This saves ~200MB in the final image.

Comment (KJ002) [reply Dockerfile:4]: Unfortunately we are using discord 1.5.1 which has dependencies that have no wheels, If we update discord.py to 1.7.3 (latest) we should be able to remove gcc as part of the build process.

**Review by Xithrius — CHANGES_REQUESTED** [review]
Problem: pipenv.

File: Dockerfile [hunk 3]
```diff
@@ -1,7 +1,4 @@
-FROM python:3.8-slim
-
-# Install dependencies for OpenCV
-RUN apt-get update && apt-get -y install libgl1 libglib2.0-0 git
```
Comment (fisher60) [inline Dockerfile:4]: Unfortunately we do still need to install git in order to utilize the version command. This line can be put back in, just without libg, only git.

**Review by Xithrius — APPROVED** [review]
Looks good.

**Review by harrisdurrani — APPROVED** [review]
after a rigorous review process, it looks good to me

**Review by fisher60 — APPROVED** [review]
Thank you very much. This all looks good.

**Pull request merged**